tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6292,6 +6305,11 @@ bool HLSLExternalSource::MatchArguments(
       }
     }
 
+    // If the intrinsic parameter has variable rows or columns but must match
+    // other argument dimensions, it will be specified in pIntrinsicArg with
+    // a special value indicating that the dimension depends on passed values.
+    // uSpecialSize stores the dimensions of the actual passed type.
```

Comment:
Here and below, I just try to better document what is actually happening here for my own and others' benefit.